accounts/abi/bind/bind.go

```diff
@@ -291,19 +292,14 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 		templateSource string
 	)
 
-	// Generate the contract template data according to contract type (precompile/non)
-	if isPrecompile {
-		if lang != LangGo {
-			return "", errors.New("only GoLang binding for precompiled contracts is supported yet")
-		}
-
-		if len(contracts) != 1 {
-			return "", errors.New("cannot generate more than 1 contract")
+	// Generate the contract template data according to  hook
+	if bindHook != nil {
+		var err error
+		data, templateSource, err = bindHook(lang, types, contracts, structs)
+		if err != nil {
+			return "", err
 		}
-		precompileType := types[0]
-		firstContract := contracts[precompileType]
-		data, templateSource = createPrecompileDataAndTemplate(firstContract, structs)
-	} else {
+	} else { // default to generate contract binding
```

Comment:
an alternative would be to pass this code in as `bindHook` for the default (`nil`) case. however there is the downside of making it more difficult to trace upstream changes. Not suggesting we make a change, just highlighting the tradeoff

Reply:
yeah, true. the reason I added this hook is to trace upstream changes more easily.